tests/integration/test_groupby.py

```diff
@@ -238,3 +238,84 @@ def test_aggregations(c):
 
     expected_df = pd.DataFrame({"max": ["a normal string"], "min": ["%_%"]})
     assert_frame_equal(df.reset_index(drop=True), expected_df)
+
```

Comment:
I like your tests, they are all very good.
However you could even increase the test cases for your new cool feature by adding a test in the comparison with SQL lite and/or postgreSQL. There are already tests for this and you probably only need to add one or two lines of SQL that call your new function (s). The benefit of having this additional tests is, that by now you only test the "definition". If we have misunderstood the definition of regr_count (for whatever reason and it is not count(y and x not null)), we would not see it here.

Reply:
Thanks for the explanation,  💯 makes sense !!

Added following function only in test_postgres.py not added in test_sqlite.py (not sure whether the SQLite supports those function)

```
def test_statistical_functions(assert_query_gives_same_result):

    # test regr_count
    assert_query_gives_same_result(
        """
        select user_id,regr_count(a,b) from df1 group by user_id
        """
    )

    assert_query_gives_same_result(
        """
        select user_id, regr_sxx(a,b) from df1 group by user_id
        """
    )

    assert_query_gives_same_result(
        """
        select user_id, regr_syy(a,b) from df1 group by user_id
        """
    )

    assert_query_gives_same_result(
        """
        select user_id,COVAR_POP(a,b) from df1 group by user_id
        """
    )

    assert_query_gives_same_result(
        """
        select user_id,COVAR_SAMP(a,b) from df1 group by user_id
        """
    )

```
Is this something you are expecting?